lib/ansible/modules/storage/netapp/na_cdot_lun.py

```diff
@@ -331,7 +331,7 @@ def apply(self):
                 property_changed = True
 
             elif self.state == 'present':
-                if not current_size == self.size:
+                if not int(current_size) == self.size:
```

Comment:
I know you didn't implement this code as it is currently constituted but I'd like to make a few suggestions that I think will leave the module in better shape than we found it.  First, can we directly compare the current_size to self.size, ```if int(current_size_str) is not self.size:```.  We can easily skim this and understand the logic.

Second, in the file, there is an assumption in the code that size is returned by the API as an int.  There are several spots where size is cast to a str despite it already being a string.  Can we add '_str' to the name of the size variable so those in the future will understand what you learned the hard way, the API returns size as an str?  

I think we can also enforce the size variable in the playbook to be an int as well.  I don't know this syntax off the top of my head though.

Reply:
comparing ints using the `is` operator is not a good idea. `is` is for identity, not equality.

> I think we can also enforce the size variable in the playbook to be an int as well. I don't know this syntax off the top of my head though.

It's already enforced to be an int, it has `type=int`.

Reply:
Man, my bad about is vs ==.  I ran a quick test with small ints and they work as expected but not correct.  Also, I was looking for type definition in the DOCUMENTATION docstring, not in the code itself.  Thank you for the contribution.  I'll go through and improve the other things I commented on; I don't want to stand in the way.